test/e2e/test_catalogue_category.py

```diff
@@ -129,6 +152,19 @@ def _post_n_catalogue_categories(test_client, number):
     )
 
 
+def _post_units(test_client):
```

Comment:
I am a little confused as to why this is returning a list of one unit dict as well as a a unit dict?

Reply:
I think it makes the tests more readable as I've used unit_dict["mm"] instead of units[0]["id"]

Reply:
Maybe for that part but I am not convinced that is the case for the rest, sorry. I don't think the return makes sense and what would happen if in future we want this test method to post more than one unit? Which one would we return as `unit_dict`? This will likely require the return to be changed which will break all the tests that do `unit_dict["mm"]`.

The two ways I can think of that this can be improved is by changing the method to only return `units` and instead of doing `_, unit_dict = _post_units(test_client)` in the tests you would do `unit = _post_units(test_client)[0]`. Alternatively, follow what @joelvdavies did in `_post_catalogue_categories` (return just the unit itself) and instead of `units, _ = _post_units(test_client)` you would do `unit_a = _post_units(test_client)`. You would then put it in a list when passing it to the `replace_unit_values_with_ids_in_properties` method.

Reply:
It should be `units_dict ` instead of `unit_dict` and if there is more then one then it would just be added into the dictionary `units_dict = {unit_mm["value"]: unit_mm["id"],unit_cm["value"]: unit_cm["id"]}`